src/components/Input.jsx

```diff
@@ -1,14 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../redux/books/bookSlice';
+import AddButton from './AddButton';
 
-const Input = () => (
-  <div>
-    <h2>Add new book</h2>
-    <form>
-      <input type="text" name="" id="" placeholder="Book Title" />
-      <input type="text" name="" id="" placeholder="Author" />
-      <button type="button">Add Book</button>
-    </form>
-  </div>
-);
+const Input = () => {
+  // <div>
+  // <h2>Add new book</h2>;
+  //   <form>
+  //     <input type="text" name="" id="" placeholder="Book Title" />
+  //     <input type="text" name="" id="" placeholder="Author" />
+  //     <button type="button">Add Book</button>
+  //   </form>
+  // </div>
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const dispatch = useDispatch();
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!title.trim() || !author.trim()) {
+      return;
+    }
+    const newBook = {
+      id: uuidv4(),
+      title,
+      author,
+    };
+    dispatch(addBook(newBook)); // Dispatch the addBook action with the newBook data
+    setTitle(''); // Clear the title input
+    setAuthor(''); // Clear the author input
+  };
+  return (
+    <div>
+      <h2>Add New Book</h2>
+      <form onSubmit={handleSubmit}>
+        {/* Add onSubmit handler */}
+        <input
+          type="text"
+          placeholder="Book title"
+          value={title}
+          onChange={(e) => setTitle(e.target.value)}
+        />
+        <input
+          type="text"
+          placeholder="Author"
+          value={author}
+          onChange={(e) => setAuthor(e.target.value)}
+        />
+        {/* <button type="submit">Add Book</button> */}
```

Comment:
- [ ] Just as I pointed out above about Following the best practices, you are not meant to leave unused code as a comment on your code base. I'll suggest you remove this as it makes your codebase look a bit a bit unprofessional. :+1: :smile: